src/main/scala/nl.knaw.dans.easy.dd2d/DatasetUpdater.scala

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (C) 2020 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.dd2d
+
+import java.nio.file.Path
+
+import gov.loc.repository.bagit.hash.StandardSupportedAlgorithms
+import nl.knaw.dans.easy.dd2d.mapping.JsonObject
+import nl.knaw.dans.lib.dataverse.DataverseInstance
+import nl.knaw.dans.lib.dataverse.model.dataset.{ MetadataBlock, MetadataBlocks, MetadataField, PrimitiveSingleValueField, toFieldMap }
+import nl.knaw.dans.lib.dataverse.model.file.FileMeta
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+
+import scala.collection.JavaConverters.{ asScalaSetConverter, mapAsScalaMapConverter }
+import scala.util.{ Failure, Success, Try }
+
+class DatasetUpdater(deposit: Deposit, metadataBlocks: MetadataBlocks, instance: DataverseInstance) extends DatasetEditor(deposit, instance) with DebugEnhancedLogging {
+  trace(deposit)
+
+  override def performEdit(): Try[PersistendId] = {
+    for {
+      _ <- dataset.updateMetadata(metadataBlocks)
+
+      pathToFileInfo <- deposit.getPathToFileInfo
+      checksumToFileInfoInDeposit <- getFilesInDeposit(pathToFileInfo)
+      checksumToFileMetaInLatestVersion <- getFilesInLatestVersion
+
+      // (old, new) checksum
+      checksumPairsToReplace <- getFilesToReplace(checksumToFileInfoInDeposit, checksumToFileMetaInLatestVersion)
+      _ <- logFilesToReplace(checksumPairsToReplace.map(_._1), checksumToFileMetaInLatestVersion)
+      _ <- replaceFiles(checksumPairsToReplace, checksumToFileMetaInLatestVersion, checksumToFileInfoInDeposit)
+
+      checksumsFilesToDelete = (checksumToFileMetaInLatestVersion.keySet diff checksumPairsToReplace.map(_._1).toSet) diff checksumToFileInfoInDeposit.keySet
```

Comment:
I think it would be easier to read if you have a method to get the intersection.  e.g. 
checksumPairsIntersection <- getIntersection(checksumToFileInfoInDeposit, checksumToFileMetaInLatestVersion)
And then checksumsFilesToDelete = checksumsToFileMetaLatestVersion.keySet diff checksumPairsIntersection.keySet

Reply:
OK, I get your point, but you got the variables mixed up.

Reply:
I am not sure what you mean, but I'll try to make the code more explicit by introducing a couple of intermediate values.